File: python/src/main/python/WordAdapter.py
from IGoogleDoc import *
from IWordDocument import *
from Format import *
from License import *
from BackImage import *


class WordAdapter(IGoogleDoc):

    def __init__(self, wordDocument):
        self.wordDocument = wordDocument

    def getFont(self):
        return self.wordDocument.get_format().getFont()

    def getStyle(self):
        return self.wordDocument.get_format()

    def getBackground(self):
        return BackgroundImage(self.wordDocument.getBackground())

    def setPermissions(self, sharing_permissions):
        return

    def getPermissions(self):
        if self.wordDocument.licenceInvalidCannotEdit():
            return 0
        return 1

File: java/src/main/java/mx/iteso/WordAdapter.java
package mx.iteso;

import mx.iteso.utils.BackgroundImage;
import mx.iteso.utils.Format;

import java.awt.*;

/**
 * Word Adapter class.
 */
public class WordAdapter implements IGoogleDoc {

    /**
     * word Document.
     */
    private IWordDocument wordDocument;

    /**
     * Word Adapter.
     *
     * @param wordDocument document.
     */
    public WordAdapter(final WordDocument wordDocument) {
        this.wordDocument = wordDocument;
    }

    /**
     * Get style.
     *
     * @return format.
     */
    public Format getStyle() {
        return this.wordDocument.getFormat();
    }

    /**
     * Get font.
     *
     * @return Font.
     */
    public Font getFont() {
        return this.wordDocument.getFormat().getFont();
    }

    /**
     * Set sharing permissions.
     *
     * @param sharingPermissions sharingPermissions.
     */
    public void setSharingPermissions(final int sharingPermissions) {
        return;
    }

    /**
     * Get background.
     *
     * @return BackgroundImage.
     */
    public BackgroundImage getBackground() {
        return new BackgroundImage(this.wordDocument.getBackground());
    }

    /**
     * Get sharing permissions.
     *
     * @return int.
     */
    public int getSharingPermissions() {
        return this.wordDocument.restrictEditIfLicenseIsInvalid() ? 0 : 1;
    }
}
